### PR TITLE
Own search event logging 

### PR DESCRIPTION
**Pull request by cbart**
Part of https://github.com/sourcegraph/sourcegraph/issues/47062

## Test plan

[Check it out on loom](https://github.com/sourcegraph/sourcegraph/pull/new/cb/add-ownership-search-telemetry)

New unit test 🌮

### CODE REVIEW

File: internal/search/job/jobutil/log_job_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				Query: q,
+			}
+			db := database.NewMockDB()
+			eventStore := &fakeEventLogStore{}
+			db.EventLogsFunc.SetDefaultReturn(eventStore)
+			ctx := actor.WithActor(context.Background(), actor.FromUser(42))
+			childJob := mockjob.NewMockJob()
+			childJob.RunFunc.SetDefaultHook(func(_ context.Context, _ job.RuntimeClients, s streaming.Sender) (*search.Alert, error) {
+				return nil, nil
+			})
```
Comment (camdencheek) [inline internal/search/job/jobutil/log_job_test.go:73]: Just an FYI that this is just explicitly encoding the default behavior. Methods on `NewMock*` will always just return the default values unless mocked. If you want to fail when an unmocked method is called, you can use `NewStrictMock*`. Additionally, for the case when you don't care about the arguments, you can achieve the same result with `childJob.RunFunc.SetDefaultReturn(nil, nil)`.

File: internal/search/job/jobutil/log_job_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			voidCollector := streaming.StreamFunc(func(ev streaming.SearchEvent) {})
```
Comment (camdencheek) [inline internal/search/job/jobutil/log_job_test.go:74]: Note to self: would probably be good to add a `NullStream` to the streaming package. This is a pretty common need.

**Review by camdencheek — APPROVED** [review]

**Review by eseliger — APPROVED** [review]
Don't know a lot about search job logging, but code looks valid. Maybe someone from search can give a final approval.

File: internal/search/job/jobutil/log_job_test.go (hunk 1 above)
Comment (cbart) [reply internal/search/job/jobutil/log_job_test.go:73]: Thanks! For some reason I was thinking it panics.

File: internal/search/job/jobutil/log_job_test.go (hunk 2 above)
Comment (cbart) [reply internal/search/job/jobutil/log_job_test.go:71]: :spinner:

File: internal/search/job/jobutil/log_job.go [hunk 3]
```diff
@@ -144,6 +144,22 @@ func (l *LogJob) logEvent(ctx context.Context, clients job.RuntimeClients, durat
 			if err != nil {
 				clients.Logger.Warn("Could not log search latency", log.Error(err))
 			}
+
+			if _, _, ok := isOwnershipSearch(q, l.inputs.Features); ok {
```
[the remark this answers is not in the document]
Comment (cbart) [reply internal/search/job/jobutil/log_job.go:148]: I don't think I understand the comment. I'd be happy to address it, but I'd also look to land this rather soon. Please let me know when you're up, and I can push a follow-up pull-request.

File: internal/search/job/jobutil/log_job_test.go (hunk 2 above)
Comment (cbart) [reply internal/search/job/jobutil/log_job_test.go:71]: https://github.com/sourcegraph/sourcegraph/pull/48764

**Pull request merged**